internal/resource/config/sp/sp_authentication_policy_contract_mapping_resource.go

```diff
@@ -0,0 +1,286 @@
+	resp.TypeName = req.ProviderTypeName + "_sp_authentication_policy_contract_mapping"
+}
+
+func (r *spAuthenticationPolicyContractMappingResource) Configure(_ context.Context, req resource.ConfigureRequest, _ *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	providerCfg := req.ProviderData.(internaltypes.ResourceConfiguration)
+	r.providerConfig = providerCfg.ProviderConfig
+	r.apiClient = providerCfg.ApiClient
+}
+
+func readSpAuthenticationPolicyContractMappingResourceResponse(ctx context.Context, r *client.ApcToSpAdapterMapping, state *spAuthenticationPolicyContractMappingResourceModel, plan spAuthenticationPolicyContractMappingResourceModel) diag.Diagnostics {
+	var diags, respDiags diag.Diagnostics
+	state.AttributeSources, respDiags = attributesources.ToState(ctx, r.AttributeSources)
+	diags.Append(respDiags...)
+	state.AttributeContractFulfillment, respDiags = attributecontractfulfillment.ToState(ctx, r.AttributeContractFulfillment)
+	diags.Append(respDiags...)
+	state.IssuanceCriteria, respDiags = issuancecriteria.ToState(ctx, r.IssuanceCriteria)
+	diags.Append(respDiags...)
+	state.SourceId = types.StringValue(r.SourceId)
+	state.TargetId = types.StringValue(r.TargetId)
+	state.Id = types.StringPointerValue(r.Id)
+	state.DefaultTargetResource = types.StringPointerValue(r.DefaultTargetResource)
+	state.LicenseConnectionGroupAssignment = types.StringPointerValue(r.LicenseConnectionGroupAssignment)
+	return diags
+}
+
+func (r *spAuthenticationPolicyContractMappingResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan spAuthenticationPolicyContractMappingResourceModel
+
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	attributeContractFulfillment := &map[string]client.AttributeFulfillmentValue{}
+	attributeContractFulfillmentErr := json.Unmarshal([]byte(internaljson.FromValue(plan.AttributeContractFulfillment, false)), attributeContractFulfillment)
+	if attributeContractFulfillmentErr != nil {
+		resp.Diagnostics.AddError("Failed to build attribute contract fulfillment request object:", attributeContractFulfillmentErr.Error())
+		return
+	}
+	createSpAuthenticationPolicyContractMappingResource := client.NewApcToSpAdapterMapping(*attributeContractFulfillment, plan.SourceId.ValueString(), plan.TargetId.ValueString())
+	err := addOptionalSpAuthenticationPolicyContractMappingResourceFields(ctx, createSpAuthenticationPolicyContractMappingResource, plan)
+	if err != nil {
+		resp.Diagnostics.AddError("Failed to add optional properties to add request for SpAuthenticationPolicyContractMappingResource", err.Error())
+		return
+	}
+	requestJson, err := createSpAuthenticationPolicyContractMappingResource.MarshalJSON()
+	if err == nil {
+		tflog.Debug(ctx, "Add request: "+string(requestJson))
```

Comment:
This and the response should change to be more like line 230 in the idp_default_urls_resource.go. Doing so will eliminate the possibility of accidental sensitive output.